tests/integration/store-test.js

```diff
@@ -209,6 +209,13 @@ test("destroying the store correctly cleans everything up", function(assert) {
   assert.equal(filterdPeopleWillDestroy.called.length, 1, 'expected filterdPeople.willDestroy to have been called once');
 });
 
+testInDebug("_setupRelationshipsForModel generates helpful error in non-prod.", function(assert) {
+  run(store, 'destroy');
+  assert.throws(() => {
+    store._setupRelationshipsForModel(null, { relationships: {} });
+  }, /Attempting to set up relationships after store has been destroyed/);
+});
```

Comment:
We should design a test scenario similar to a real-world scenario that would have caused this to happen.  I'm especially interested in such a test because:

1. It may be that we need to adjust our waiters https://github.com/emberjs/data/pull/5422
2. _setupRelationshipsForModel should still occur in the same run-loop as whatever scheduled it. This indicates to me that this either might be the wrong location to assert, or that we could improve the message to tell people exactly what they need to do to resolve the error.

Reply:
Hi @runspired.  I agree that we should eventually dig more into the root issue, but this PR at least make the situation 1 step better - Someone encountering this in dev or test will know more about what's going on and fix it faster.

I unfortunately won't have time to dig deeper for a while, so maybe we could ship this as-is?

Reply:
@danwenzel if you can jot down some notes for me about roughly the setup you encountered in the wild, and give me commit access to your fork, I will continue to improve this :)

Reply:
Hi @runspired , sorry for the delay.

I can't remember the details of my setup, but this gives a pretty good example of the issue: https://github.com/samselikoff/ember-cli-mirage/issues/1220.

I've given you access to my fork.  Let me know if you need anything else